### PR TITLE
Implement the last design pattern

### PR DESCRIPTION
**Pull request by AxelF44**
I've Implemented the last desgin: Builder

So I use it for Camera because we've already the Factory for Light and Primitives so that's why I decided to do that !

### CODE REVIEW

File: include/Camera/Camera.hpp [hunk 1]
```diff
@@ -39,6 +39,14 @@ class Raytracer::Camera {
          */
         Camera(Math::Point3D origin, Rectangle3D screen);
 
+        /**
+         * @brief Construct a new Camera object by another one
```
Comment (theophile-jr) [inline include/Camera/Camera.hpp:43]: "." add

Comment (Njord201) [inline include/Camera/Camera.hpp:43]:
```suggestion
         * @brief Construct a new Camera object by another one.
```

File: src/Core/Camera/CameraBuilder.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
+/*
+** EPITECH PROJECT, 2024
+** Camera.cpp
```
Comment (theophile-jr) [inline src/Core/Camera/CameraBuilder.cpp:3]: CameraBuilder.cpp

File: include/Camera/CameraBuilder.hpp [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+         *
+         */
+        ~CameraBuilder() = default;
+
+        /**
+         * @brief Set the Origin object.
+         *
+         * @param origin Origin to set
+         */
+        CameraBuilder& setOrigin(Math::Point3D origin);
```
Comment (Njord201) [inline include/Camera/CameraBuilder.hpp:36]: Describe the return value.

**Review by theophile-jr — CHANGES_REQUESTED** [review]

File: include/Camera/CameraBuilder.hpp [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+        /**
+         * @brief Set the Screen object.
+         *
+         * @param screen Screen to set
+         */
+        CameraBuilder& setScreen(Raytracer::Rectangle3D screen);
```
Comment (Njord201) [inline include/Camera/CameraBuilder.hpp:43]: Describe the return value in the documentation

File: include/Camera/CameraBuilder.hpp [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+        /**
+         * @brief Set the Fov object.
+         *
+         * @param fov Fov to set
+         */
+        CameraBuilder& setFov(double fov);
```
Comment (Njord201) [inline include/Camera/CameraBuilder.hpp:50]: Describe the return value of the function in the documentation

File: include/Camera/CameraBuilder.hpp [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+        /**
+         * @brief Set the Rotation object.
+         *
+         * @param rotation Rotation Vector to set
+         */
+        CameraBuilder& setRotation(Math::Vector3D rotation);
```
Comment (Njord201) [inline include/Camera/CameraBuilder.hpp:57]: Describe the return value of the function in the documentation

File: include/Camera/CameraBuilder.hpp [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+        /**
+         * @brief Set the Resolution object.
+         *
+         * @param width of image
+         * @param height of image
+         */
+        CameraBuilder& setResolution(double width, double height);
```
Comment (Njord201) [inline include/Camera/CameraBuilder.hpp:65]: Describe the return value of the function in the documentation

File: include/Camera/CameraBuilder.hpp [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+        /**
+         * @brief Build the Camera object.
+         * 
+         * @return Camera to build
+         */
+        Camera build(void);
+
+    private:
+        Camera camera;
```
Comment (Njord201) [inline include/Camera/CameraBuilder.hpp:75]:
```suggestion
        Camera _camera;
```

**Review by Njord201 — CHANGES_REQUESTED** [review]

**Review by theophile-jr — APPROVED** [review]

**Pull request merged**